code/backend/batch/utilities/search/search_handler_base.py

```diff
@@ -12,10 +12,15 @@ def __init__(self, env_helper: EnvHelper):
         self.env_helper = env_helper
         self.search_client = self.create_search_client()
 
-    def search_with_facets(self, query: str, facets: list[str]):
+    def search_with_facets(self, query: str, facets: list[str], facet_count: int):
         if self.search_client is None:
             return None
-        return self.search_client.search(query, facets=facets)
+
+        # Construct facet parameter based on facet_count
+        facet_param = f"{facets[0]},count:{facet_count}"
```

Comment:
Sorry I should have picked up on this sooner. 

`facets` is a list, but we're only getting the first one, meaning any extra are ignored. However, I don't think we actually ever pass in more than one, so I'm happy for the fix to either be:
- Include all `facets` in the `facets_param`
- Change the `facets` argument to be `facet: str`

Reply:
@adamdougal We have made the changes. can you please review and approve.